enhancements/sig-architecture/82-addon-template/README.md

```diff
@@ -0,0 +1,407 @@
+The addon can be enabled for the corresponding managed cluster by creating a `ManagedClusterAddon`.
+
+This simplifies the addon development process, but it still requires developers to implement the interface defined in
+the addon-framework via code, and the development process is still slightly complicated. so we hope to explore a more
+straightforward addon development way.
+
+### Goals
+
+- Provide a new addon development model, developers can define an addon YAML template and some parameters to describe
+  how the addon agent will be installed on the managed cluster and how to register it to the hub, developers do not need
+  to deploy an addon-manager on the hub cluster to manage the addon agent.
+- Make the addon agent deployment and registration configurable, modifying the deployment and registration logic does
+  not need to recompile the addon image.
+
+### Non-Goals
+
+- The addon agent image is still required.
+- For complicated addon which has special RBAC requirement etc would still need to use addon-framework.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an addon developer, I hope that I can focus on addon business development. After preparing the addon agent image, I
+only need to define some configurations to deploy and register my addon agent.
+
+#### Story 2
+
+As an addon developer, when the installation and registration of the addon agent changes, I hope that the change can
+take effect by only modifying some configurations without changing the code
+
+## Design Details
+
+The design is based on [Add-on Lifecycle](https://github.com/open-cluster-management-io/enhancements/tree/main/enhancements/sig-architecture/81-addon-lifecycle).
+
+### Overview
+
+The main idea is to provide a new API (CRD), `AddonTemplate`, which is used to describe how to deploy the addon agent
+and how to register the addon. Then add a new addon template controller to the addon-manager component on the hub
+cluster to process the CRD in order to deploy and register the addon.
+
+### API changes
+
+Introduce a new cluster-scoped API `AddonTemplate` which is used to describe how to deploy the addon agent and how to
+register the addon to the hub cluster. And it is needed to reference the `AddonTemplate` by the "spec.supportedConfigs"
+field of `ClusterManagementAddon` to indicate that this addon is using a template.
+
+```golang
+// AddOnTemplateSpec defines the template of an addon agent which will be deployed on managed clusters.
+type AddOnTemplateSpec struct {
+	// AddonName represents the name of the addon which the template belongs to
+	// +kubebuilder:validation:Required
+	// +required
+	AddonName string `json:"addonName"`
+
+	// AgentManifests represents the kubernetes resources of the addon agent to be deployed on a managed cluster.
+	// +kubebuilder:validation:Required
+	// +required
+	AgentManifests []Manifest `json:"agentManifests"`
```

Comment:
I would nest this in case you end up needing to add more out of ManifestWorkSpec .  Perhaps `template` similar to how we nest a PodTemplateSpec for workload resources.  You don't have to pre-emptively include the entire resource, but I would structure this to allow for expansion to include it.